### PR TITLE
Integration of rest and sql modules

### PR DESCRIPTION
**Pull request by stasya72008**

### CODE REVIEW

File: rest/app.py [hunk 1]
```diff
@@ -1,78 +1,75 @@
...
+    return render_template('index.html'), 204
 
 
 @app.route('/users/<int:user_id>', methods=['GET'])
 def get_user(user_id):
-    for user in users:
-        if user['id'] == user_id:
-            return json.dumps(user)
+    user = sql_users.get_user(user_id)
+    if user:
```
Comment (denyskulykov) [inline rest/app.py]: Please use more shotly construction like
return json() if user else 404

File: rest/app.py [hunk 2]
```diff
@@ -1,78 +1,75 @@
...
-    # ToDo(stasya): add request to database
-    user_tasks = list()
-    for task in tasks:
-        if task['user_id'] == user_id:
-            # SQL request for the given period
-            user_tasks.append(task)
-
-    return json.dumps(user_tasks)
+def get_user_tasks(user_id):
+    period = request.args
```
Comment (denyskulykov) [inline rest/app.py]: Please add docstring with example of using this method
Which  parameters are expected?

File: rest/app.py [hunk 3]
```diff
@@ -1,78 +1,75 @@
...
+    month = None
+    day = None
+    if period:
+        year = period['year']
+        if 'month' in period:
```
Comment (denyskulykov) [inline rest/app.py]: You can use the construction like:
_month = period.get('month', None)_
It lets you skip setting value before and _if month in ..._

File: rest/app.py [hunk 4]
```diff
@@ -1,78 +1,75 @@
...
 
 @app.route('/tasks/<int:task_id>', methods=['GET'])
-def get_task(task_id):
-    for task in tasks:
-        if task['id'] == task_id:
-            return json.dumps(task)
+def get_task_by_id(task_id):
+    task = sql_tasks.get_task(task_id)
+    if task:
+        return json.dumps(task)
```
Comment (denyskulykov) [inline rest/app.py]: the same like _get_user()_

File: rest/app.py [hunk 5]
```diff
@@ -81,27 +78,33 @@ def add_task():
...
+    name = ''
+    status = None
+    calendar_date = None
     data = request.json
-    for task in tasks:
-        if task['id'] == task_id:
-            task.update(data)
-            return json.dumps(task)
-    return not_found(404)
+    if 'name' in data:
```
Comment (denyskulykov) [inline rest/app.py]: the same like: _get_user_tasks()_

File: rest/templates/index.html [hunk 6]
```diff
@@ -5,14 +5,43 @@
...
-    {% for user in users %}
-    <h2>Hello, {{ user.username }}!</h2>
-    {% endfor %}
-
-    {% for task in tasks %}
-    <div><p>{{ task.date }}: <b>{{ task.name }}</b></p></div>
-    <div><p> ... </p></div>
-    {% endfor %}
-
+<div>
```
Comment (denyskulykov) [inline rest/templates/index.html:8]: =)

File: web_app/rest_client/client.py [hunk 7]
```diff
@@ -23,15 +23,14 @@ def get_task_by_id(task_id):
...
     return json.loads(resp.text)
 
 
-def create_task(user_id, task_name, calendar_date=''):
-    task = template.task
+def create_task(user_id, task_name, calendar_date=None):
+    task = dict()
     # ToDo(den) move user_id to init class rest_client after
     # https://github.com/stasya72008/highway-to-hell/projects/2#card-11180275
     task['user_id'] = user_id
```
Comment (denyskulykov) [inline web_app/rest_client/client.py]: Please drop the line _task['id'] = random.randint(10, 1000000)_

File: rest/app.py (hunk 1 above)
Comment (stasya72008) [reply rest/app.py]: What construction?)

File: rest/app.py [hunk 8]
```diff
@@ -1,107 +1,105 @@
...
 
 
 @app.route('/tasks', methods=['POST'])
 def add_task():
     # ToDo(stasya) Add json validation
     # https://github.com/stasya72008/highway-to-hell/projects/2#card-11155844
     task = json.loads(request.json)
-    tasks.append(task)
-    return render_template('index.html'), 201
+    user_id = task['user_id']
```
Comment (denyskulykov) [inline rest/app.py]: I think You can drop lines _user_id = task['user-id']_
and send this parameter directly  
sql_tasks.add_task(task['user-id'], ...)

File: rest/app.py [hunk 9]
```diff
@@ -1,107 +1,105 @@
...
-    return json.dumps(users)
+    return json.dumps(sql_users.get_all_users())
 
 
 @app.route('/users', methods=['POST'])
 def add_user():
     user = json.loads(request.json)
-    users.append(user)
-    return render_template('index.html'), 201
+    user_name = user['name']
```
Comment (denyskulykov) [inline rest/app.py]: The same like in **add_task()**

File: web_app/rest_client/client.py [hunk 10]
```diff
@@ -23,21 +23,19 @@ def get_task_by_id(task_id):
...
```
Comment (denyskulykov) [inline web_app/rest_client/client.py]: You can set dict like 
task = {
'user_id': = user_id, ... }

**Pull request merged**